### PR TITLE
Require node 16

### PR DESCRIPTION
**Pull request by danield9tqh**
## Summary
The CLI is built to run on node version 16. We have run into one [issue](https://github.com/iron-fish/ironfish/issues/2376) that arose from incompatible node version. We should require users to be using the correct node version so we do not run into these versioning errors.

## Testing Plan
Local testing

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish-cli/bin/run [hunk 1]
```diff
@@ -7,6 +7,11 @@ if (process.platform !== 'win32') {
   require('segfault-handler').registerHandler('segfault.log')
 }
 
+if(process.versions.node.split('.')[0] !== '16') {
+  console.log('NodeJS version ' +  process.versions.node + ' is not compatible. Must have node v16.x installed. You can install using nvm: https://github.com/nvm-sh/nvm')
```
Comment (NullSoldier) [inline ironfish-cli/bin/run:11]: Can we just link to node 16? NVM might be too hard for non technical people.

**Review by hughy — COMMENTED** [review]
Could we enforce this on install?

e.g., https://stackoverflow.com/a/29349773

**Comment by NullSoldier** [comment]
If you do the package strategy, does that work for execution too? I think we want it on both install / execution ideally. In the case that you're using NVM and forget to switch node versions.

File: ironfish-cli/bin/run (hunk 1 above)
Comment (danield9tqh) [reply ironfish-cli/bin/run:11]: Yea good call

**Comment by NullSoldier** [comment]
@hughy: @danield9tqh  is going to look into preventing this on install as well since there's more questions about if that works properly. We'll merge this in and expect a PR open for fixing users installing it using the wrong version. We can stop them a lot sooner! Especially because it'll cause modules to get built under the correct node version.

**Review by NullSoldier — APPROVED** [review]
Tested manually, this works well.

**Pull request merged**